src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
 export interface Figure {
-
+  shape: string;
+  color: string;
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape = 'triangle';
+
+  color: string;
+
+  sides: [number, number, number];
+
+  constructor(color: string, ...sides: [number, number, number]) {
+    if (sides.some((side) => side <= 0) || this.isNotTriangle(sides)) {
+      throw new Error('Invalid triangle');
+    }
+    this.color = color;
+    this.sides = sides;
+  }
+
+  getArea(): number {
+    const [a, b, c] = this.sides;
+    const s = (a + b + c) / 2;
+
+    return Number((Math.sqrt(s * (s - a) * (s - b) * (s - c))).toFixed(2));
+  }
+
+  private isNotTriangle = (sides: [number, number, number]): boolean => {
+    const [a, b, c] = sides;
 
+    return a + b <= c || a + c <= b || b + c <= a;
+  };
 }
 
-export class Circle {
+export class Circle implements Figure {
+  shape = 'circle';
 
+  color: string;
+
+  radius: number;
+
+  constructor(color: string, radius: number) {
+    if (radius <= 0) {
+      throw new Error('Invalid circle');
+    }
+    this.color = color;
+    this.radius = radius;
+  }
+
```

Comment:
try to simplify it using the typescript shorthand constructor
https://dev.to/satansdeer/typescript-constructor-shorthand-3ibd